com.baidu.NetDisk.yml

```diff
@@ -0,0 +1,65 @@
+id: com.baidu.NetDisk
+base: org.electronjs.Electron2.BaseApp
+base-version: '19.08'
+runtime: org.freedesktop.Platform
+runtime-version: '19.08'
+sdk: org.freedesktop.Sdk
+tags:
+  - proprietary
+command: baidunetdisk
+separate-locales: false
+build-options:
+  no-debuginfo: true
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --device=dri
+  - --talk-name=org.freedesktop.Notifications
+  - --persist=baidunetdisk
+  - --persist=.accelerate
+  - --persist=.pki
+  - --filesystem=xdg-download
+  - --filesystem=~/baidunetdiskdownload
+  - --env=GTK_IM_MODULE=xim
+cleanup:
+  - "*.a"
+  - "*.la"
+  - /include
+  - /lib/pkgconfig
+  - /lib/cmake
+modules:
+  - name: baidunetdisk
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 apply_extra.sh /app/bin/apply_extra
+      - install -Dm644 ${FLATPAK_ID}.appdata.xml -t /app/share/appdata/
+      - install -Dm755 /usr/bin/desktop-file-edit -t /app/bin/
+      - install -Dm755 /usr/bin/ar -t /app/bin/
+      - cp -a /usr/lib/x86_64-linux-gnu/libbfd* /app/lib
+      - ln -s ../extra/baidunetdisk/baidunetdisk /app/bin/baidunetdisk
+    sources:
+      - type: script
+        dest-filename: apply_extra.sh
+        commands:
+          - ar x baidunetdisk.deb data.tar.xz
+          - tar fx data.tar.xz
+          - rm data.tar.xz baidunetdisk.deb
+          - mv opt/* .
+          - rmdir opt
+          - mv usr export
+          - rm -fr export/share/doc
+          - desktop-file-edit --set-key="Exec" --set-value="baidunetdisk %U" export/share/applications/baidunetdisk.desktop
+          - desktop-file-edit --set-key="Icon" --set-value="com.baidu.NetDisk" export/share/applications/baidunetdisk.desktop
+          - desktop-file-edit --set-key="X-Flatpak-RenamedFrom" --set-value="baidunetdisk.desktop;" export/share/applications/baidunetdisk.desktop
+          - rename "baidunetdisk." "com.baidu.NetDisk." export/share/{icons/hicolor/*/*,applications}/baidunetdisk.*
+
+      - type: file
+        path: com.baidu.NetDisk.appdata.xml
```

Comment:
New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html